concordium-std/src/types.rs

```diff
@@ -1,11 +1,100 @@
-use crate::num::NonZeroU32;
+use crate::{num::NonZeroU32, HasContractStateEntry, HasContractStateLL};
+use std::{cell::RefCell, marker::PhantomData, rc::Rc};
+
+#[derive(Debug)]
+pub struct StateMap<K, V, S>
+where
+    S: HasContractStateLL, {
+    pub(crate) _marker_key:   PhantomData<K>,
+    pub(crate) _marker_value: PhantomData<V>,
```

Comment:
I am not sure whether this will be helpful or detrimental, but just adding it here so we can come back to it.
It might be better to have `PhantomData<*const K>` and the same for `V` instead.

See https://doc.rust-lang.org/std/marker/struct.PhantomData.html#ownership-and-the-drop-check